www/js/equation.js

```diff
@@ -39,23 +39,45 @@ define([
 				});
 				return equation;
 			}
-			// is there any significance for the following object? else delete
-			this.mapVariableNodeNames = {};
-			// console.log("            parse: ", expr);
 
 			if(params.nameToId){
 				var nodeList = []; //this holds new node ids and variable objects
 				var variableList = []; // this holds the variable list
 				var connections = []; // this holds the input list
+				var dynamicList = []; //this list holds the prior nodes which are to be created further in controller
+
 				array.forEach(expressions, function(expr){
 					variableList = variableList.concat(expr.variables());
+					var currentPriorList = expr.priors();
 					array.forEach(expr.variables(), function(variable){
 						//This is the case where node names have to be converted to ids
 						//This situation arises from equationDoneHandler
 						//In such situation convert name to id
 						var nodeId = subModel.getNodeIDByName(variable);
 						if(nodeId){
 							expr.substitute(variable,nodeId);
```

Comment:
Finally, we decided the current flow is correct because
1. if a and prior(a) are in equation expr.substitute replaces a with id 
2. further substitutePrior will replace prior(a) with id_initial
both cases are handled in the flow